src/PortingAssistantExtensionClientShared/Utils/AwsUtils.cs

```diff
@@ -136,5 +157,51 @@ TelemetryConfiguration telemetryConfiguration
                 return false;
             }
         }
+
+        public static async Task<bool> ValidateProfileAsync()
+        {
+            if (UserSettings.Instance.AWSProfileName != null || UserSettings.Instance.EnabledDefaultCredentials)
+            {
+                var chain = new CredentialProfileStoreChain();
+                AWSCredentials awsCredentials;
+
+                if (UserSettings.Instance.EnabledDefaultCredentials)
+                {
+                    awsCredentials = FallbackCredentialsFactory.GetCredentials();
+                    if (awsCredentials == null)
+                    {
+                        await NotificationUtils.ShowInfoBarAsync(PAGlobalService.Instance.AsyncServiceProvider, "AWS Credentials associated with Porting Assistant for .NET Expired. Please refresh credentials.");
+                        return false;
+                    }
+                }
+                else
+                {
+                    var profileName = UserSettings.Instance.AWSProfileName;
+                    if (!chain.TryGetAWSCredentials(profileName, out awsCredentials))
+                    {
+                        await NotificationUtils.ShowInfoBarAsync(PAGlobalService.Instance.AsyncServiceProvider, "AWS Credentials associated with Porting Assistant for .NET Expired. Please refresh credentials.");
+                        return false;
+                    }
+                }
```

Comment:
just a thought. What if we wrapped the "getcreds" logic in a method, and then used the success of that method instead of the "(UserSettings.Instance.AWSProfileName != null || UserSettings.Instance.EnabledDefaultCredentials)". It would also allow us to consolidate the expired credentials method of the creds returned is null?